openmc/tallies.py

```diff
@@ -3522,6 +3527,18 @@ def _create_mesh_subelements(self, root_element):
                         root_element.append(f.mesh.to_xml_element())
                         already_written.add(f.mesh)
 
+    def _create_filter_subelements(self, root_element):
+        already_written = dict()
+        for tally in self:
+            for f in tally.filters:
+                if f not in already_written:
+                    root_element.append(f.to_xml_element())
+                    already_written[f] = f.id
+                else:
+                    # Set the IDs of identical filters with different
+                    # user-defined IDs to the same value
+                    f.id = already_written[f]
```

Comment:
Yes, that's right. I was a little wary of modifying the IDs, could this cause problems anywhere? Also, is there a reason `__hash__` and `__eq__` generally aren't consistent?